ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
+    # + return - `()` if successful or else `error`
+    isolated remote function changePassword(string userId, byte[] newPassword) returns error? {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, USER, userId, PASSWORD]);
+        record{} payload = {"NewPassword" : check string:fromBytes(newPassword)};
+        return check self.salesforceClient->post(path, payload);
     }
 
-    # Creates records based on relevant object type sent with json record.
+    # Returns a list of actions and their details
     #
-    # + sObjectName - SObject name value
-    # + recordPayload - JSON record to be inserted
-    # + return - Created entity ID if successful or else `sfdc:Error`
-    #
-    # # Deprecated
-    # This function is deprecated as it expects a json payload.
-    # Use the new and improved `create(string sObjectName, record{} recordPayload)` function instead.
-    @display {label: "Create Record"}
-    @deprecated
-    isolated remote function createRecord(@display {label: "SObject Name"} string sObjectName,
-                                        @display {label: "Record Payload"} json recordPayload)
-                                        returns @display {label: "Created Entity ID"} string|Error {
-        http:Request req = new;
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName]);
-        req.setJsonPayload(recordPayload);
-        json|http:ClientError response = self.salesforceClient->post(path, req);
-        if response is json {
-            json|error resultId = response.id;
-            if resultId is json {
-                return resultId.toString();
-            } else {
-                return error Error(resultId.message());
-            }
-        } else {
-            return checkAndSetErrorDetail(response);
-        }
-    }
-
-    private isolated function createUtil(@display {label: "SObject Name"} string sObjectName,
-                                        @display {label: "Record Payload"} json recordPayload)
-                                        returns @display {label: "Created Entity ID"} string|Error {
-        http:Request req = new;
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName]);
-        req.setJsonPayload(recordPayload);
-        json|http:ClientError response = self.salesforceClient->post(path, req);
-        if response is json {
-            json|error resultId = response.id;
-            if resultId is json {
-                return resultId.toString();
-            } else {
-                return error Error(resultId.message());
-            }
+    # + sObjectName - SObject reference
+    # + return - `QuickAction[]` if successful or else `error`
+    isolated remote function getQuickActions(string sObjectName) returns QuickAction[]|error {
+        string path = utils:prepareUrl([API_BASE_PATH, QUICK_ACTIONS]);
+        http:Response response = check self.salesforceClient->get(path);
```

Comment:
```suggestion
        QuickAction[] actions = check self.salesforceClient->get(path);
```

Can we do direct data binding?

Reply:
Updated in [update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/311/commits/cd6ae536defa51bd247b84742f1c480c2cf80ab1)